src/Materials/materialHelper.ts

```diff
@@ -340,11 +341,21 @@ export class MaterialHelper {
         if (mesh && mesh.receiveShadows && scene.shadowsEnabled && light.shadowEnabled) {
             var shadowGenerator = light.getShadowGenerator();
             if (shadowGenerator) {
-                const shadowMap = shadowGenerator.getShadowMap();
-                if (shadowMap) {
-                    if (shadowMap.renderList && shadowMap.renderList.length > 0) {
-                        state.shadowEnabled = true;
-                        shadowGenerator.prepareDefines(defines, lightIndex);
+                if (shadowGenerator.useCSM) {

```

Comment:
This looks pretty similar I guess we could facto a bit

Reply:
Done!
```
var shadowMap;
if (shadowGenerator.useCSM) {
	shadowMap = shadowGenerator.getAllCSMs()[0];
} else {
	shadowMap = shadowGenerator.getShadowMap();
}
if (shadowMap) {
	if (shadowMap.renderList && shadowMap.renderList.length > 0) {
	state.shadowEnabled = true;
	shadowGenerator.prepareDefines(defines, lightIndex);
	}
}
```